core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -108,6 +110,19 @@ class BrokerTopicMetrics(name: Option[String]) extends KafkaMetricsGroup {
   val failedFetchRequestRate = newMeter("FailedFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalProduceRequestRate = newMeter("TotalProduceRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
   val totalFetchRequestRate = newMeter("TotalFetchRequestsPerSec", "requests", TimeUnit.SECONDS, tags)
+
+  val partitions = new ConcurrentHashMap[Int, Partition]
```

Comment:
(or equivalently just an atomic int that increments on add and decrements on remove)